src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,103 @@
-import React from 'react';
+import React, { useState, Dispatch, useEffect } from 'react';
 import './TodoList.scss';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
+interface Props {
+  todos: Todo[];
+  setSelectedUserId: Dispatch<number>,
+}
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  setSelectedUserId,
+}) => {
+  const [isComplete, setIsComplete] = useState('show all');
+  const [filterByTitle, setFilterByTitle] = useState<string>('');
+  const [visibleTodos, setVisibleTodos] = useState(todos);
+
+  useEffect(() => {
+    setVisibleTodos(todos.filter(todo => {
+      const titleToLower = todo.title.toLowerCase();
+      const filterByTitleToLower = filterByTitle.toLowerCase();
+
+      switch (isComplete) {
+        case 'show all':
+          return titleToLower.includes(filterByTitleToLower);
+
+        case 'not completed':
+          return titleToLower.includes(filterByTitleToLower)
+          && todo.completed === false;
+
+        case 'completed':
+          return titleToLower.includes(filterByTitleToLower)
+          && todo.completed === true;
+
+        default:
+          return todo;
```

Comment:
better just throw exception there smth like "not existing type of filter exception"